e2e/test/MessageReceiveE2ETests.cs

```diff
@@ -114,7 +114,14 @@ public static async Task VerifyReceivedC2DMessageAsync(Client.TransportType tran
                 Client.Message receivedMessage = null;
 
                 _log.WriteLine($"Receiving messages for device {deviceId}.");
-                receivedMessage = await dc.ReceiveAsync(TimeSpan.FromSeconds(1)).ConfigureAwait(false);
+                if (transport == Client.TransportType.Http1)
```

Comment:
It might be worth adding a note here as to why we don't pass the TimeSpan to the Http call

Reply:
On HTTP timeout is not supported, we throw an exception.

Reply:
Added a comment